components/server/ee/src/user/eligibility-service.ts

```diff
@@ -93,22 +93,26 @@ export class EligibilityService {
             return { student: false, faculty: false };
         }
 
+        const logCtx = { userId: user.id };
         try {
             const rawResponse = await fetch("https://education.github.com/api/user", {
                 headers: {
                     "Authorization": `token ${token}`,
                     "faculty-check-preview": "true"
                 }
             });
+            if (!rawResponse.ok) {
```

Comment:
nice catch! I believe in that case it doesn't make sense to proceed with `await rawResponse.text()` as it will throw.